### PR TITLE
[Bugfix] Fix Chromedriver download in case of lack of specified version

### PR DESCRIPTION
**Pull request by HannaTarasevich**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

This pull request is aimed to fix the bug https://github.com/webdriverio/webdriverio/issues/10860 available version of Chrome for desktop is 115.0.5790.171, but the stable Chromedriver version is 115.0.5790.170. Previously WDIO checked if Chromedriver is installed based on the Chrome version installed. But the current Chrome logic is that a different build version for stable Chrome and Chromedriver is possible, and this leads to a WDIO error in attempting to download Chromedriver 115.0.5790.171.

Current changes include the following Chromedriver flow:

1. If there is a downloaded specified Chromedriver version (build number based on installed Chrome)
2. If not downloaded - check if it is possible to download this specified Chromedriver version
3. If not possible - check if the stable Chromedriver version has already been downloaded
4. If not downloaded - it downloads the stable Chromedriver version

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/webdriver/src/driver/utils.ts [hunk 1]
```diff
@@ -114,6 +115,49 @@ export async function setupChrome(caps: Capabilities.Capabilities, cacheDir: str
...
+    const installOptions: InstallOptions & { unpack?: true } = {
+        ...chromedriverOptions,
+        cacheDir,
+        platform,
+        buildId,
+        browser: Browser.CHROMEDRIVER,
+        unpack: true,
+        downloadProgressCallback: (downloadedBytes: number, totalBytes: number) => downloadProgressCallback('Chromedriver', downloadedBytes, totalBytes)
+    }
+    if (!(await hasChromedriverInstalled(chromedriverBinaryPath))) {
```
Comment (christian-bromann) [inline packages/webdriver/src/driver/utils.ts:137]: No need for `hasChromedriverInstalled` to be a function.
```suggestion
    const hasChromedriverInstalled = await fsp.access(chromedriverBinaryPath)
        .then(() => true, () => false)
    // Check if specified Chromedriver version is installed
    const installOptions: InstallOptions & { unpack?: true } = {
        ...chromedriverOptions,
        cacheDir,
        platform,
        buildId,
        browser: Browser.CHROMEDRIVER,
        unpack: true,
        downloadProgressCallback: (downloadedBytes: number, totalBytes: number) => downloadProgressCallback('Chromedriver', downloadedBytes, totalBytes)
    }
    if (!hasChromedriverInstalled) {
```

File: packages/webdriver/src/driver/utils.ts [hunk 2]
```diff
@@ -114,6 +115,49 @@ export async function setupChrome(caps: Capabilities.Capabilities, cacheDir: str
...
+        // Check if specified version of Chromedriver is available for download
+        const isCombinationAvailable = await canDownload(installOptions)
+        // If specified chromedriver version is not installed and there is no specified version to download, work with the latest stable version
+        if (!isCombinationAvailable){
+            const stableChromedriverBuildId = await resolveBuildId(Browser.CHROMEDRIVER, platform, ChromeReleaseChannel.STABLE)
```
Comment (christian-bromann) [inline packages/webdriver/src/driver/utils.ts:142]: This assumes that the user wants to test on a stable release. What happens this error appears for an older version?

**Review by christian-bromann — COMMENTED** [review]
Thanks for taking a stab to solve this bug. My only concern is that we should make sure to respect the Chrome version we have set up or identified prior setting up Chromedriver. For example what happens the user sets a Chrome version which has no Chromedriver correspondent?

**Comment by HannaTarasevich** [comment]
> Thanks for taking a stab to solve this bug. My only concern is that we should make sure to respect the Chrome version we have set up or identified prior setting up Chromedriver. For example what happens the user sets a Chrome version which has no Chromedriver correspondent?

Great point, thanks! I'll continue work on this point

**Comment by christian-bromann** [comment]
@HannaTarasevich due to the amount of bug reports that came in I tried to come up with a new approach, see #10902

**Pull request closed without merging**